modules/storage/cosmos-db/main.bicep

```
@@ -1,15 +1,11 @@
-@description('Deployment region name. Default is the location of the resource group.')
-param location string = resourceGroup().location
-
 @description('The bakend API type of Cosmos DB database account. The API selection cannot be changed after account creation. Possible values: "cassandra", "gremlin", "mongodb", "sql", "table".')
 @allowed([ 'cassandra', 'gremlin', 'mongodb', 'sql', 'table' ])
 param backendApi string = 'sql'
 
-@description('Prefix of Cosmos DB Resource Name. Not used if name is provided.')
-param prefix string = { cassandra: 'coscas', gremlin: 'cosgrm', mongodb: 'cosmon', sql: 'cosmos', table: 'costab' }[backendApi]
-
 @description('The name of the Cosmos DB account. Character limit: 3-44, valid characters: lowercase letters, numbers, and hyphens. It must me unique across Azure.')
-param name string = take('${prefix}-${uniqueString(resourceGroup().id, location)}', 44)
+@maxLength(44)
+@minLength(3)
+param name string = uniqueString(resourceGroup().id, resourceGroup().location)
```

```suggestion
param name string = take('${prefix}-${uniqueString(resourceGroup().id, location)}', 44)
```

keep the take to make sure the maxLength is kept in check

---

`uniqueString` only produced 13 characters, so the default string will always be 20 chars long. There is no need for `take(0`